### PR TITLE
issue #719 solved

### PR DESCRIPTION
**Pull request by ArchishmanSengupta**
Added the following as discussed in issue #719 
Prerequisites
How to Report Bugs
How to Request Enhancements
Setting up the local environment
Forking repository 
 Clone repository 
Setting up remote 
Pushing your changes
Creating a pull request

Thank you

### CODE REVIEW

**Comment by ArchishmanSengupta** [comment]
@joernu76 please review the documentation. Thank you

File: docs/development.rst [hunk 1]
```diff
@@ -1,6 +1,91 @@
 .. _development:
 
+1. System requirement :
 
+Any system with basic configuration.
+Operating System : Any (Windows / Linux / Mac).
+
+2. Software requirement :
+Updated browser
```
Comment (ReimarBauer) [inline docs/development.rst]: what is the reason for an updated browser?

File: docs/development.rst [hunk 2]
```diff
@@ -250,3 +335,29 @@ Publish on Conda Forge
 * send a pull request
 * maintainer will merge if there is no error
 
+
+============================
+Pushing your changes:
+============================
+
+1. Now you have made the changes , tested them, and built them. so now its time to push them.
```
Comment (ReimarBauer) [inline docs/development.rst]: typo " , "
typo "its" / "it's"

File: docs/development.rst [hunk 3]
```diff
@@ -1,6 +1,91 @@
 .. _development:
 
+1. System requirement :
```
Comment (ReimarBauer) [inline docs/development.rst]: should that be  `1. System requirement:` or is that blank before the colon needed, or do we need the colon at all?

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
a few questions, thx for the description

File: docs/development.rst (hunk 1 above)
Comment (ArchishmanSengupta) [reply docs/development.rst]: It was a mistake. Will delete this.

File: docs/development.rst (hunk 2 above)
Comment (ArchishmanSengupta) [reply docs/development.rst]: Noted.

File: docs/development.rst (hunk 3 above)
Comment (ArchishmanSengupta) [reply docs/development.rst]: I will remove the colon. 👍🏻

File: docs/development.rst [hunk 4]
```diff
@@ -1,6 +1,91 @@
...
 
+Any system with basic configuration.
+Operating System : Any (Windows / Linux / Mac).
```
Comment (joernu76) [inline docs/development.rst]: To have a proper enumerated list, this needs to be indented. See https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html and test the syntax with tools such as rst2pdf or rst2latex

File: docs/development.rst [hunk 5]
```diff
@@ -1,6 +1,91 @@
...
+python
+
+============================
+Setting Up local Environement
+============================
+
+============================
+Forking the Repo
+============================
+1.Firstly you have to make your own copy of project. For that you have to fork the repository. You can find the fork button on the top-right side of the browser window.
```
Comment (joernu76) [inline docs/development.rst]: There must be a space " " after the "1.", i.e., "1. Firstly". Same below.

File: docs/development.rst [hunk 6]
```diff
@@ -1,6 +1,91 @@
...
+
+2.Kindly wait till it gets forked.
+
+3.After that copy will look like <your-user-name>/MSS forked from Open-MSS/MSS.
+
+============================
+Cloning the Repo
+============================
+
+1.Now you have your own copy of project. Here you have to start your work.
```
Comment (joernu76) [inline docs/development.rst]: Add spaces behind the numbers.

File: docs/development.rst [hunk 7]
```diff
@@ -1,6 +1,91 @@
...
+
+2.Go to desired location on your computer where you want to set-up the project.
+
+3.Right click there and click on git bash. A terminal window will pop up
```
Comment (joernu76) [inline docs/development.rst]: This looks like a Windows/Mac thing? I cannot do that on my Debian-Mate desktop. Please keep it OS-agnostic.

File: docs/development.rst [hunk 8]
```diff
@@ -1,6 +1,91 @@
...
+
+4.Type the command git clone <your-fork-url>.git and hit enter.
```
Comment (joernu76) [inline docs/development.rst]: Git gives an easy reference to your repo and instructions how to clone. Please advise people to use that (The big green button with CODE).

File: docs/development.rst [hunk 9]
```diff
@@ -1,6 +1,91 @@
...
+
+5.Wait for few seconds till the project gets copied
```
Comment (joernu76) [inline docs/development.rst]: I think that is obvious as git gives a nice progress report.

File: docs/development.rst [hunk 10]
```diff
@@ -1,6 +1,91 @@
...
+
+============================
+Setting up remote :
+============================
+
+1.Now you have to set up remote repositories
```
Comment (joernu76) [inline docs/development.rst]: Add spaces after enumeration,

File: docs/development.rst [hunk 11]
```diff
@@ -1,6 +1,91 @@
...
+
+2.Type git remote -v in terminal to list remote connections to your repo.
+
+3. It will show something like this:
+
+origin  https://github.com/<your-user-name>/MSS.git (fetch)
```
Comment (joernu76) [inline docs/development.rst]: Indentation!

File: docs/development.rst [hunk 12]
```diff
@@ -1,6 +1,91 @@
...
+
+origin  https://github.com/<your-user-name>/MSS.git (push)
+
+4. Now type the command git remote add upstream https://github.com/Open-MSS/MSS.git this will set upstream as main directory
+
+5. Again type in command git remote -v to check if remote has been set up correctly
+
+6. It should show something like this :
+
+origin  https://github.com/<your-user-name>/MSS.git (fetch)
```
Comment (joernu76) [inline docs/development.rst]: Add indentation!

File: docs/development.rst [hunk 13]
```diff
@@ -40,8 +125,8 @@ mss repository needs a different folder, e.g. workspace/mss. Avoid to mix data a
 
 Some of used packages are in the conda-forge channel located, so we have to add this channel to the default::
 
-  $ conda config --add channels conda-forge
-  $ conda config --add channels defaults
+$ conda config --add channels conda-forge
```
Comment (joernu76) [inline docs/development.rst]: You need to add some indentation or similar (verbatim text) to keep this from becoming a normal paragraph.

File: docs/development.rst [hunk 14]
```diff
@@ -250,3 +335,29 @@ Publish on Conda Forge
...
+
+============================
+Pushing your changes:
+============================
+
+1. Now you have made the changes , tested them, and built them. so now its time to push them.
+
+2. Goto your terminal and type git status and hit enter, this will show your changes from the files
+3. Then type in git add and hit enter, this will add all the files to staging area
+4. Commit the changes by git commit -m "<message-describing-your-change>" and hit enter.
```
Comment (joernu76) [inline docs/development.rst]: Highlight the git command and it's parameters with backquotation as code example (backquotes: ``text`` for code samples.)

File: docs/development.rst [hunk 15]
```diff
@@ -1,6 +1,91 @@
...
+
+2. Software requirement :
+Updated browser
+python
+
+3. Skill set :
+Knowledge of git & github.
+python
+
+============================
```
Comment (joernu76) [inline docs/development.rst:18]: The other ReST files use only lines below the heading. Please keep it consistent.

File: docs/development.rst [hunk 16]
```diff
@@ -1,6 +1,91 @@
...
+
+5.Wait for few seconds till the project gets copied
+
+============================
+Setting up remote :
+============================
+
+1.Now you have to set up remote repositories
+
+2.Type git remote -v in terminal to list remote connections to your repo.
```
Comment (joernu76) [inline docs/development.rst]: Highlight the command (backquotes: ``text`` for code samples.)

File: docs/development.rst [hunk 17]
```diff
@@ -1,6 +1,91 @@
...
+
+3. It will show something like this:
+
+origin  https://github.com/<your-user-name>/MSS.git (fetch)
+
+origin  https://github.com/<your-user-name>/MSS.git (push)
+
+4. Now type the command git remote add upstream https://github.com/Open-MSS/MSS.git this will set upstream as main directory
+
+5. Again type in command git remote -v to check if remote has been set up correctly
```
Comment (joernu76) [inline docs/development.rst:65]: Highlight the command (backquotes: ``text`` for code samples.)

**Review by joernu76 — CHANGES_REQUESTED** [review]
First, sort out the syntax and test the file locally. Second, I really like step-by-step instructions, but this is indeed difficult to do right. I would advise to link more to "professional" documentation such as the https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository page for cloning a repository.

File: docs/development.rst (hunk 4 above)
Comment (ReimarBauer) [reply docs/development.rst]: or by sphinx locally, all needed is in the development requirements

Comment (ArchishmanSengupta) [reply docs/development.rst]: Thanks for this. Needed this

**Comment by ReimarBauer** [comment]
I do convert this to a draft state until it is continued

**Comment by ArchishmanSengupta** [comment]
> I do convert this to a draft state until it is continued

hey @ReimarBauer I will be resolving and doing all the changes. i just need some time. Hope you understand.

**Comment by ReimarBauer** [comment]
@ArchishmanSengupta How do you do?

**Comment by ArchishmanSengupta** [comment]
> @ArchishmanSengupta How do you do?

hey, I have done the necessary changes, now should I make a new pr or go ahead with ```resolve conflicts``` with the whole refactoring of the rst file?

File: docs/development.rst [hunk 18]
```diff
@@ -256,3 +348,29 @@ Publish on Conda Forge
 * send a pull request
 * maintainer will merge if there is no error
 
+
+============================
+Pushing your changes:
+============================
+
+1. Now you have made the changes , tested them, and built them. so now it's time to push them.
```
Comment (ReimarBauer) [inline docs/development.rst]: " , " one blank after a ","

File: docs/development.rst (hunk 15 above)
Comment (ReimarBauer) [reply docs/development.rst:18]: this is a kind of "chapter" style, or? But I also think we should not overuse this. And may be later do a refactoring where it is better used.

File: docs/development.rst (hunk 18 above)
Comment (ReimarBauer) [reply docs/development.rst]: s/so now/So now/

**Comment by ArchishmanSengupta** [comment]
will I close the PR as I have to update my local repo, so that I can contribute a new PR?

**Review by joernu76 — APPROVED** [review]

**Comment by ReimarBauer** [comment]
Just a note. This is a good start for giving the information. In a further step this needs a refactoring by moving all development related parts into the section Development.

This has to be done before we can release 5.0. I file for this a blocking issue.

These entries have to be sorted :)

menu

**Review by ReimarBauer — APPROVED** [review]

**Pull request merged**